CMakeLists.txt

```diff
@@ -205,6 +207,11 @@ else()
     message(STATUS "Not using Custom.cmake file.")
 endif()
 
+if(LIBSAMPLERATE_LIBRARY AND LIBSAMPLERATE_INCLUDE_DIRECTORY)
+  set(LIBSAMPLERATE_FOUND TRUE)
```

Comment:
This should be set in the find module: https://github.com/csound/csound/blob/a1580f9cdf331c35dceb486f4231871ce0b00266/cmake/Modules/FindEIGEN3.cmake#L66

So you import it along with the rest of the set or unset vars. You should avoid using `include_` family of functions and use `target_` only to avoid including these paths for all targets.

Reply:
I did that initially, the problem I encountered, is that I am defining the location of `LIBSAMPLERATE_INCLUDE_DIRECTORY` in my custom.cmake file. Since it says in the comment where the custom.cmake file is imported https://github.com/csound/csound/blob/develop/CMakeLists.txt#L197-L199 that the idea of it is to override variables from cmake after find_/locate_ operations are done, then it means that the value of `LIBSAMPLERATE_INCLUDE_DIRECTORY` wont exist until later.
I am aware that this is a terrible excuse. I'm just worried about needing to resort to -DVar.. flags for this, which is bit annoying.
The correct solution, in my opinion, is that custom.cmake will be placed at the top (or very close to it) and that way, the developers don't get access to "cheat-mode", because if cmake doesn't resolve the locations correctly when the custom file is evaluated at the beginning, it will also translate to non reproduce-able situations when we are trying to debug issues from users.
With custom.cmake at the top, I'd be totally open for having a second custom cmake file, called something like, overrides.cmake, that gets called after all variables are set.

The second point you made, preferring `target_` over `include_`, totally, I will fix :)

Reply:
I'm thinking to just not have a _FOUND variable and check for both vars existing, it willsuffice until we can rethink (or not) the order of custom cmake import.

Reply:
fixed

Reply:
Afaik the idea of custom.cmake is mostly for setting feature flags for the app. I don't think you should be overriding find_ module vars this way. You can add a path hint maybe.